glean-core/src/upload/mod.rs

```diff
@@ -575,6 +592,13 @@ impl PingUploadManager {
                     }
                 }
 
+                {
+                    let mut in_flight = self.in_flight.write().unwrap();
+                    let success_id = self.upload_metrics.sending_success.start_sync();
+                    let failure_id = self.upload_metrics.sending_failure.start_sync();
```

Comment:
Musing: I wonder if we should have safety notices for synchronous I/O. Like "Sync I/O: Acceptable because it's on the uploader thread which itself does waiting I/O on the network" or something. Might not be a problem until someone finds us on a perf profile.